lib/charms/postgresql_k8s/v1/postgresql.py

```diff
@@ -918,20 +916,23 @@ def set_up_database(self, temp_location: Optional[str] = None) -> None:
           -- Don't execute on replicas.
           IF NOT pg_is_in_recovery() THEN
             -- Load the current authorisation rules.
-            PERFORM TRUE FROM pg_tables WHERE schemaname LIKE 'pg_temp_%' AND tablename = 'pg_hba';
-            IF FOUND THEN
-                DROP TABLE pg_hba;
+            SELECT nspname INTO temp_schema FROM pg_namespace WHERE  oid = pg_my_temp_schema();
+            IF temp_schema != '' THEN
+                PERFORM TRUE FROM pg_tables WHERE schemaname = temp_schema AND tablename = 'pg_hba';
+                IF FOUND THEN
+                    DROP TABLE pg_hba;
+                END IF;
+                PERFORM TRUE FROM pg_tables WHERE schemaname = temp_schema AND tablename = 'relation_users';
+                IF FOUND THEN
+                    DROP TABLE relation_users;
+                END IF;
```

Comment:
There may be another `pg_temp_*` temporary schema created by another connection, so we need to check if the temporary tables (`pg_hba` and `relations`) exist in the temporary schema related to the current connection. Otherwise, the `DROP TABLE` statement will fail.